Attributes/Troops/Logic/attribute_5.gd

```diff
@@ -0,0 +1,29 @@
+extends TroopAttribute
+
+# Logic for splash attribute
+
+func build_action() -> Action:
+    var action: Action = Action.new()
+    action.name = attribute.name
+    action.description = attribute.description
+    action.setup(parent.game, on_attack)
+    return action
+
+#loops to all tiles around defender and applies half damage
+
+func on_attack(defender: Unit):
+    var defender_pos = defender.pos
+    var splash_tiles = _get_surrounding(defender_pos, 1)
+
+    for tile in splash_tiles:
+        var unit_at_tile = game.board.units[tile.x][tile.y]
+        apply_splash_damage(unit_at_tile)
```

Comment:
Before applying splash damage, check if board.units[tile.x][tile.y] == null and continue if so.